core/src/main/java/com/arcrobotics/ftclib/command/button/Trigger.java

```diff
@@ -283,6 +283,64 @@ public Trigger toggleWhenActive(final Command command) {
         return toggleWhenActive(command, true);
     }
 
+    /**
+     * Toggles between two commands when the trigger becomes active (commadOne then commandTwo
+     * then commandOne).
+     *
+     * @param commandOne the command to toggle
+     * @param commandTwo the command to be toggled
+     * @param interruptible whether the commands are interruptible
+     * @return this trigger, so calls can be chained
+     */
+    public Trigger toggleWhenActive(final Command commandOne, final Command commandTwo, boolean interruptible) {
+        CommandScheduler.getInstance().addButton(new Runnable() {
+            private boolean m_pressedLast = get();
+            @Override
+            public void run() {
+                boolean pressed = get();
+                if (!m_pressedLast && pressed) {
+                    if (commandOne.isScheduled()) {
+                        commandOne.cancel();
+                        commandTwo.schedule(interruptible);
+                    } else if(commandTwo.isScheduled()){
+                        commandTwo.cancel();
+                        commandOne.schedule(interruptible);
+                    } else {
+                        commandOne.schedule(interruptible);
+                    }
```

Comment:
This wouldn't work if the command is cancelled after one iteration (e.g. an `InstantCommand`) or is cancelled before the next active state. I suggest using a boolean here that contains the state called `m_firstCommandActive` outside of the `run()` method. It should be instantiated to `false`. You can then limit this to just a simple if-else. Make sure your spacing is consistent. Do Ctrl+Alt+L to auto-format.

Below is how this should be done:
```java
if (m_firstCommandActive) {
  if (commandOne.isScheduled()) {
    commandOne.cancel();
  }
  commandTwo.schedule(interruptible);
} else {
  if (commandTwo.isScheduled()) {
    commandTwo.cancel();
  }
  commandOne.schedule(interruptible);
}
m_firstCommandActive = !m_firstCommandActive;